src/main/java/com/esri/samples/displayinformation/sketch_on_map/SketchOnMapSample.java

```diff
@@ -0,0 +1,415 @@
+        graphicsOverlay.clearSelection();
+        disableButtons();
+        // set text to inform the user the sketch is disabled
+        stopSketchButton.setText("Sketch is disabled");
+        saveButton.setText("Save sketch to graphics overlay");
+
+        if (!graphicsOverlay.getGraphics().isEmpty()) {
+          editButton.setText("Select graphic to edit");
+          clearButton.setDisable(false);
+        }
+
+        // allow graphics to be selected after stopSketch button is used.
+        selectGraphic();
+      });
+
+      // use sketch editor to edit the geometry of the selected graphic
+      editButton.setOnAction(event -> {
+        stopSketchButton.setDisable(false);
+        saveButton.setDisable(true);
+        saveButton.setText("Save edits to graphics overlay");
+        stopSketchButton.setText("Stop sketching");
+        editButton.setText("Edit Sketch (active)");
+
+        // if the graphics overlay contains graphics, select the first graphic
+        // and start the sketch editor based on that graphic's geometry
+        if (!graphicsOverlay.getSelectedGraphics().isEmpty()) {
+          graphic = graphicsOverlay.getSelectedGraphics().get(0);
+          sketchEditor.start(graphic.getGeometry());
+        }
+      });
+
+      // clear the graphics overlay, and disable the clear button
+      clearButton.setOnAction(event -> {
+        graphicsOverlay.getGraphics().clear();
+        sketchEditor.stop();
+        disableButtons();
+      });
+
+      // add a listener for when sketch geometry is changed
+      sketchEditor.addGeometryChangedListener(SketchGeometryChangedListener -> {
+
+        // if sketch is valid, enable save and stop sketch buttons
+        if (sketchEditor.isSketchValid()) {
+          saveButton.setText("Save sketch to graphics overlay");
+          saveButton.setDisable(false);
+          stopSketchButton.setDisable(false);
+        } else {saveButton.setDisable(true);}
+
+        // if the sketch editor can undo, enable the undo button otherwise disable it
+        if (sketchEditor.canUndo()) {
+          undoButton.setDisable(false);
+        } else {
+          undoButton.setDisable(true);
+        }
+
+        // if the sketch editor can redo, enable the redo button otherwise disable it
+        if (sketchEditor.canRedo()) {
+          redoButton.setDisable(false);
+        } else {
+          redoButton.setDisable(true);
+        }
+      });
+
+      // create a control panel
+      VBox controlsVBox = new VBox(6);
+      controlsVBox.setBackground(new Background(new BackgroundFill(Paint.valueOf("rgba(0, 0, 0, 0.3)"), CornerRadii.EMPTY, Insets.EMPTY)));
+      controlsVBox.setPadding(new Insets(10));
+      controlsVBox.setMaxSize(255, 110);
+      controlsVBox.getStyleClass().add("panel-region");
+
+      // create a flow pane for placing buttons side by side within the control box
+      FlowPane flowPaneUndoRedo = new FlowPane(Orientation.HORIZONTAL, 70, 10, undoButton, redoButton);
```

Comment:
A FlowPane is really only needed when you want the layout to be responsive to layout width/height. Like if you wanted to change the layout going from a desktop sized window to a mobile sized window. I think here a simple VBox containing an HBox for the Undo/Redo buttons would suffice.